File: backend/schemas/economico/Costos.py
// src/services/economicoCostosService.ts
export type CostoEconomico = {
  id: string;
  fecha: string;
  categoria: "Fijo" | "Variable" | "Indirecto" | "Otro";
  concepto: string;
  monto: number;
  responsable: string;
  descripcion: string;
};

const STORAGE_KEY = "economico_costos";

function getAll(): CostoEconomico[] {
  const raw = localStorage.getItem(STORAGE_KEY);
  return raw ? JSON.parse(raw) : [];
}

function saveAll(lista: CostoEconomico[]) {
  localStorage.setItem(STORAGE_KEY, JSON.stringify(lista));
}

export function getCostos() {
  return getAll();
}

export function addCosto(c: Omit<CostoEconomico, "id">) {
  const nuevo: CostoEconomico = { ...c, id: crypto.randomUUID() };
  const lista = getAll();
  lista.push(nuevo);
  saveAll(lista);
  return nuevo;
}

export function updateCosto(id: string, c: Omit<CostoEconomico, "id">) {
  const lista = getAll();
  const idx = lista.findIndex(x => x.id === id);
  if (idx >= 0) {
    lista[idx] = { ...c, id };
    saveAll(lista);
    return lista[idx];
  }
  throw new Error("Registro no encontrado");
}

export function deleteCosto(id: string) {
  const lista = getAll().filter(x => x.id !== id);
  saveAll(lista);
}

File: frontend/src/services/economico/Costos.ts
// src/services/Costos.ts
import API_BASE from "../../config/api";

export type CostoEconomico = {
  id: number;
  fecha: string;
  categoria: "Fijo" | "Variable" | "Indirecto" | "Otro";
  concepto: string;
  monto: number;
  responsable: string;
  descripcion: string;
};

export type NuevoCostoEconomico = Omit<CostoEconomico, "id">;

const EMPRESA_ID = 1;
const GRANJA_ID = 1;

function mapApiToCosto(api: any): CostoEconomico {
  return {
    id: api.id,
    fecha: api.fecha,
    categoria: api.categoria,
    concepto: api.concepto,
    monto: api.monto,
    responsable: api.responsable,
    descripcion: api.descripcion ?? "",
  };
}

export async function getCostos(): Promise<CostoEconomico[]> {
  const res = await fetch(
    `${API_BASE}/economico/costos/?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`
  );
  if (!res.ok) {
    throw new Error("Error al obtener costos económicos");
  }
  const data = await res.json();
  return data.map(mapApiToCosto);
}

export async function addCosto(
  c: NuevoCostoEconomico
): Promise<CostoEconomico> {
  const payload = {
    empresa_id: EMPRESA_ID,
    granja_id: GRANJA_ID,
    ...c,
  };
  const res = await fetch(`${API_BASE}/economico/costos/`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) {
    throw new Error("Error al crear costo económico");
  }
  const data = await res.json();
  return mapApiToCosto(data);
}

export async function updateCosto(
  id: number,
  c: NuevoCostoEconomico
): Promise<CostoEconomico> {
  const res = await fetch(
    `${API_BASE}/economico/costos/${id}?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`,
    {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(c),
    }
  );
  if (!res.ok) {
    throw new Error("Error al actualizar costo económico");
  }
  const data = await res.json();
  return mapApiToCosto(data);
}

export async function deleteCosto(id: number): Promise<void> {
  const res = await fetch(
    `${API_BASE}/economico/costos/${id}?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`,
    { method: "DELETE" }
  );
  if (!res.ok) {
    throw new Error("Error al eliminar costo económico");
  }
}
